activesupport/lib/active_support/inflector/inflections.rb

```diff
@@ -76,8 +76,7 @@ def self.instance_or_fallback(locale)
       attr_reader :acronyms_camelize_regex, :acronyms_underscore_regex # :nodoc:
 
       def initialize
-        @plurals, @singulars, @uncountables, @humans, @acronyms = [], [], Uncountables.new, [], {}
-        define_acronym_regex_patterns
+        clear(:all)
```

Comment:
@odlp I agree that clear might be a bit strange when used for initialization.
Looking at other Rails classes we could stick to a plain initialization.

```suggestion
def initialize
  @plurals      = []
  @singulars    = []
  @humans       = []
  @uncountables = Uncountables.new
  @acronyms     = {}
end
```

And since there are attr_readers for these ivars we could do:

```ruby
def clear(scope = :all)
  case scope
  when :all
    clear(:arconyms)
    plurals.clear
    singulars.clear
    uncountables.clear
    humans.clear
  when :acronyms
    acronyms.clear
    define_acronym_regex_patterns
  when :plurals, :singulars, :humans, :uncountables
    public_send(scope).clear
  end
end
```

Reply:
Sorry for bikeshedding on your first PR 😄

Reply:
I gave it a try using `#clear` on each ivar, but this actually leads to a bunch of tests failing because we end up mutating the original translations (preserved in the `setup` method of the test).

I tried adding using the `#initialize_dup` utility method to get a proper isolated duplicate:

```diff
diff --git a/activesupport/lib/active_support/inflector/inflections.rb b/activesupport/lib/active_support/inflector/inflections.rb
index 4d909bb220..9372a88a70 100644
--- a/activesupport/lib/active_support/inflector/inflections.rb
+++ b/activesupport/lib/active_support/inflector/inflections.rb
@@ -54,6 +54,11 @@ def uncountable?(str)
           @regex_array.any? { |regex| regex.match? str }
         end
 
+        def clear
+          @regex_array.clear
+          super
+        end
+
         private
           def to_regex(string)
             /\b#{::Regexp.escape(string)}\Z/i
@@ -86,6 +91,7 @@ def initialize_dup(orig) # :nodoc:
           instance_variable_set("@#{scope}", orig.public_send(scope).dup)
         end
         define_acronym_regex_patterns
+        self
       end
 
       # Specifies a new acronym. An acronym must be specified as it will appear
@@ -235,12 +241,10 @@ def clear(scope = :all)
           clear(:uncountables)
           clear(:humans)
         when :acronyms
-          @acronyms = {}
+          @acronyms.clear
           define_acronym_regex_patterns
-        when :uncountables
-          @uncountables = Uncountables.new
-        when :plurals, :singulars, :humans
-          instance_variable_set "@#{scope}", []
+        when :plurals, :singulars, :humans, :uncountables
+          public_send(scope).clear
         end
       end
 
diff --git a/activesupport/test/inflector_test.rb b/activesupport/test/inflector_test.rb
index 7814326e00..027b77c06a 100644
--- a/activesupport/test/inflector_test.rb
+++ b/activesupport/test/inflector_test.rb
@@ -16,8 +16,9 @@ def setup
     # This helper is implemented by setting @__instance__ because in some tests
     # there are module functions that access ActiveSupport::Inflector.inflections,
     # so we need to replace the singleton itself.
-    @original_inflections = ActiveSupport::Inflector::Inflections.instance_variable_get(:@__instance__)[:en]
-    ActiveSupport::Inflector::Inflections.instance_variable_set(:@__instance__, en: @original_inflections.dup)
+    @original_inflections = ActiveSupport::Inflector::Inflections.instance(:en)
+    dup = ActiveSupport::Inflector::Inflections.new.send(:initialize_dup, @original_inflections)
+    ActiveSupport::Inflector::Inflections.instance_variable_set(:@__instance__, en: dup)
   end
 
   def teardown
```

But there are quite a failures, and it's unclear what state is still shared:

```
466 runs, 1179 assertions, 144 failures, 0 errors, 0 skips
```

For the moment I'm just going to put the `#initialize` method back as it was, but leave `#clear` assigning new instances.

Reply:
Thanks for giving it a try!